File: PP/Lista1/replicate.ml
(*
    O(n)
    a' * int -> a' list
*)
let rec replicate (x,y) = if y <= 0 then [] else x :: replicate (x, y-1);;

let rec print_list xs =
    match xs with
    | [] -> ()
    | e::l -> print_string e ; print_string " " ; print_list l;;

print_list (replicate ("la", 3));;

File: PP/Lista1/replicate.scala
object replicate {
    def main(args: Array[String]) : Unit = {
        def replicate[A](x : A, y : Int) : List[A] =
            (x, y) match {
                case (_, 0) => List()
                case (x, y) => x :: replicate(x, y-1)
            }
        println(replicate("la", 3))
    }
}